packages/node/src/events/install/controller.ts

```diff
@@ -67,14 +67,14 @@ function createAppInstanceFromAppInstanceInfo(
   return new AppInstance(
     channel.multisigAddress,
     // TODO: generate ephemeral app-specific keys
-    channel.multisigOwners,
+    channel.getSigningKeysFor(channel.numInstalledApps - 1),
     proposedAppInstanceInfo.timeout.toNumber(),
     appInterface,
     terms,
     // TODO: pass correct value when virtual app support gets added
     false,
     // TODO: this should be thread-safe
-    channel.numInstalledApps + 1,
+    channel.numInstalledApps - 1,
```

Comment:
Why -1?

If it's the first AppInstance being installed, 0 is already taken by the ETHFreeBalance, so shouldn't this be +1?

Reply:
This is an issue that will get fixed in `liam/virtualapp`. Basically this function is creating a "representation" of an `AppInstance`. However, the `channel` variable is the `StateChannel` _after_ the "real" `AppInstance` has already been installed. 

This helper function will be removed in my other PR because it is confusing and unnecessary.